onnxruntime/core/providers/dml/DmlExecutionProvider/src/FusedGraphKernel.cpp

```diff
@@ -13,6 +13,26 @@ namespace Dml
 {
     class FusedGraphKernel : public onnxruntime::OpKernel
     {
+    private:
+        struct ReusedCommandListState
+        {
+            // Re-usable command list, supporting descriptor heap, and DML binding table to update that heap.
+            ComPtr<ID3D12GraphicsCommandList> graphicsCommandList;
+            ComPtr<ID3D12CommandAllocator> commandAllocator;
+            ComPtr<ID3D12DescriptorHeap> heap;
+            ComPtr<IDMLBindingTable> bindingTable;
+
+            // Bindings from previous executions of a re-used command list
+            mutable std::vector<uint64_t> inputBindingAllocIds;
+            mutable std::vector<uint64_t> outputBindingAllocIds;
+            mutable uint64_t tempBindingAllocId = 0;
+
+            // Fence tracking the status of the command list's last execution, and whether its descriptor heap
+            // can safely be updated.
+            mutable ComPtr<ID3D12Fence> fence;
+            mutable uint64_t completionValue = 0;
```

Comment:
Are those `mutable` keywords still necessary? `mutable std::deque<std::unique_ptr<ReusedCommandListState>> m_reusedCommandLists;` is already marked as mutable below.

Reply:
I believe it makes the deque mutable.
If I take it off the build errors when modifying `m_reusedCommandLists` Line 172
`m_reusedCommandLists.push_front(BuildReusableCommandList());`

Reply:
This is usually the case when a const method attempts to change data. Should the method NOT be const?

Reply:
The EP more generally does not support concurrent execution.  The kernel needs to maintain caches, and those caches are opaque to ORT.  I don't believe there are much better alternatives in this case, and it's following a pre-existing pattern.